namesys/namesys.go

```diff
@@ -1,59 +1,95 @@
 package namesys
 
 import (
+	"strings"
+
 	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
 	ci "github.com/ipfs/go-ipfs/p2p/crypto"
 	path "github.com/ipfs/go-ipfs/path"
 	routing "github.com/ipfs/go-ipfs/routing"
 )
 
-// ipnsNameSystem implements IPNS naming.
+// ipfs NameSystem implements generic IPFS naming.
```

Comment:
this is not about godoc or the ipfs implementations-- it's about how we describe the system that resolves:

```
/ipns/Qm..../...
/ipns/foo.com/...
```
It comes to bear on the end users, who should be required to learn the least possible amount. Yeah i corroborate my earlier comments-- this should stay defined as IPNS. it's less pure than inventing another abstraction, but the description:

> "ipns supports sfs-style pki based names, and other resolvers (dns + ...)."

is pure enough and gives end users less to learn. Until a day in which `fsroot` is trivial to deploy on the world, let's keep it as it's been: IPNS is the _name system_. it resolves SFS-style PKI names, domains, and more.